internal/contour/handler_test.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contour
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventHandlerNotRequireLeaderElection(t *testing.T) {
```

Comment:
curious what people think of these tests for the various components, settled on this after spiking through another option which i wonder what people think of

- new object as thin wrapper around controller-runtime manager
- mainly for testing, can test by mocking controller-runtime manager (see controller tests for example)
- instead of `manager.Add(Runnable)` method, have something like `manager.Add(Runnable, requireLeaderElection bool)`

decided against this for a few reasons, mainly bc it ended up being a v thin wrapper (need to expose a lot of things from the base controller-runtime manager) and puts more logic in the `main` package, rather than closer to the components themselves which should have the knowledge about whether they need leader election or not